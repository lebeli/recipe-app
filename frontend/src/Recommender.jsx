import React, { Component } from "react";
import RecommenderImage from "./RecommenderImage";
import RecommenderHeader from "./RecommenderHeader";
import RecommenderReloadButton from "./RecommenderReloadButton";
import wolfImage from "./resources/wolf.jpg";
import LasagneImage from "./resources/lasagne.jpg";
import "./Recommender.scss";

class Recommender extends Component {
  constructor(props) {
    super(props);
    this.state = {
      image: LasagneImage
    };

    this.changeImage = this.changeImage.bind(this);
  }

  changeImage() {
    this.setState({
      image: wolfImage
    });
  }

  render() {
    return (
      <div className="Recommender" id="Container">
        <div className="RelativeParent">
          <RecommenderHeader />
          <RecommenderReloadButton onClick={this.changeImage} />
          <RecommenderImage
            image={this.state.image}
            onClick={this.props.goToDetails}
          />
        </div>
      </div>
    );
  }
}

export default Recommender;
