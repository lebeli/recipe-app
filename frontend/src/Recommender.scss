@import './styles/styles.scss';

#Container {
  height: calc(100vh - 276px);

  .RelativeParent {
    position: relative;
    top: 0;
    bottom: 100px;
    width: 100%;
    height: -webkit-fill-available;
    display: flex;
    justify-content: center;
    overflow: hidden;

    .TeaserImage {
      height: 100%;
      width: auto;
    }

    .Header {
      background: rgba(255, 255, 255, 0.5);
      display: flex;
      width: 100%;
      height: 10%;
      justify-content: center;
      align-items: center;
      position: absolute;

      .HeaderContent {
        margin: 5px;
        width: 100%;
        text-align: center;
        display: flex;
        align-items: center;
        justify-content: center;
      }
    }
  }
}

#ReloadFab {
  background: #EB672C;
  color: white;
  position: absolute;
  bottom: 20px;
}
