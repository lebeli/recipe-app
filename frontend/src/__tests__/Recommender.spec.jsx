import React from "react";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import Recommender from "../Recommender";

describe("Recommender", () => {
  test("should render elementes", () => {
    const wrapper = mount(<Recommender />);
    expect(wrapper.find(".RecommenderHeader"));
    expect(wrapper.find(".RecommenderImage"));
    expect(wrapper.find(".RecommenderReloadButton"));
  });

  test("renders correctly", () => {
    const tree = renderer.create(<Recommender />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
