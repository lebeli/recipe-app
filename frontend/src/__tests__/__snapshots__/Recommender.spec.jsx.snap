// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recommender renders correctly 1`] = `
<div
  className="Recommender"
  id="Container"
>
  <div
    className="RelativeParent"
  >
    <div
      className="Header"
    >
      <div
        className="HeaderContent"
      />
      <div
        className="HeaderContent"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
          <path
            d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
          />
        </svg>
      </div>
    </div>
    <button
      aria-label="reload"
      className="MuiButtonBase-root MuiFab-root"
      disabled={false}
      id="ReloadFab"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8z"
          />
        </svg>
      </span>
    </button>
    <div
      className="TeaserImage"
    >
      <img
        alt="finished dish"
      />
    </div>
  </div>
</div>
`;
